### DISCUSSION TITLE
Looking for a proven! & working! thermald profile for Surface Pro 7 i5

### DISCUSSION THREAD

**Discussion by macschaff** [linux-surface/linux-surface#558] [Q&A]
Hello,

First off I am relatively new to Linux and this is my first try to get Linux on a Microsoft Surface device running. I have a lot of experience when it comes to Windows but not Linux. Therefore I never before worked with thermald. So please take it easy on me. 
 
I am struggling for days now to get thermald working properly on my Microsoft Surface Pro 7 i5 with Ubuntu 21.04. Whatever configuration (thermal-conf.xml + thermal-cpu-cdev-order.xml) I tried from this website or others did not work. 
Whenever I use a CPU demanding software like a game the CPU rather soon (within 5 minutes) reaches a temperature where it emergency throttles down to 200MHz for a longer period. When I use 'stress -c 8' the CPU reaches 80 degrees Celsius within about 15 seconds even though thermald is running and seems to work (for what I can read out of the debug output as the newb I am).

Log file of a session running 'stress -c 4': 
[thermald.log](https://github.com/linux-surface/linux-surface/files/7117949/thermald.log)

Config files:
https://github.com/linux-surface/linux-surface/blob/master/contrib/thermald/thermal-conf.xml
(I set the trip point to 55000 instead of the original 65000 because with that the temp went over 70 degrees Celsius very fast. I don't want the CPU to get hotter than 70 degrees Celsius)
https://github.com/linux-surface/linux-surface/blob/master/contrib/thermald/thermal-cpu-cdev-order.xml

I am therefore looking for a proven and working thermald profile (thermal-conf.xml + 
thermal-cpu-cdev-order.xml) for the Surface Pro 7 i5 that will keep the CPU from going into a thermal emergency shutdown.

Any help would be highly appreciable.

Regards
Marcus

**Comment by astejko** [comment]
Hey, just took a quick look at your files, it looks like you are trying to cool your system using rapl_controller, which doesn't seem to exist on your device, to check which cooling devices you have, you can run `cat /sys/class/thermal/cooling_device*/type`

Let me know if you have any questions on how to set it up

**Comment by raisen** [comment]
@macschaff were you able to get a config file for the SP7 that works? I'm facing the same issue.

**Comment by Yeshey** [comment]
Refer to [this answer](https://github.com/linux-surface/linux-surface/issues/221#issuecomment-693334512), they suggest a working config file.
